xapian-core/include/xapian/weight.h

```diff
@@ -372,6 +395,10 @@ class XAPIAN_VISIBILITY_DEFAULT Weight {
     Xapian::termcount get_wdf_upper_bound() const {
 	return wdf_upper_bound_;
     }
+
+    static void parameter_error(const char * param, const std::string & scheme);
+
+    bool double_param(const char ** p, double * ptr_val) const;
```

Comment:
It's not great to add internal stuff like this such that it has to appear in the public API header - it means we can't change the calling signature of these methods within a release series.

Neither seems to use the context of a `Weight` object - could they both be made `static` and added to `Weight::Internal` instead?